### PR TITLE
Feature/private route finished

### PR DESCRIPTION
**Pull request by mattcanavan**
## Finished privateRoute and rewrote Success component for JWT

**Description:**
Finished private route. Test it out!
Rewrote Success component to handle jwt.

### CODE REVIEW

File: src/components/authentication/Success.js [hunk 1]
```diff
@@ -15,4 +29,10 @@ const Success = (props) => {
   );
 };
 
-export default connect(null, { success })(Success);
+const mapStateToProps = (state) => {
```
Comment (mxhs) [inline src/components/authentication/Success.js:35]: nice catch here.

File: src/store/actions/index.js [hunk 2]
```diff
@@ -8,26 +8,6 @@ axios.defaults.withCredentials = true;
...
-      },
-    })
-    .then((response) => {
-      if (response.data.user.track === null) {
-        history.push('/onboarding');
-      } else {
-        history.push('/');
-      }
-      toast.success('Welcome to the Lambda Alumni Network!');
-    })
```
Comment (mxhs) [inline src/store/actions/index.js:27]: The welcome message I thought was kind of nice - didn't want to keep it?

File: src/utils/privateRoute.js [hunk 3]
```diff
@@ -1,15 +1,36 @@
...
     <Route
       {...rest}
       render={(props) => {
-        if (localStorage.getItem('token')) {
+        if (activeSession) {
           return <Component {...props} />;
         } else {
-          return <Redirect to="/welcome" />;
+          <Redirect to="/welcome" />;
         }
```
Comment (mxhs) [inline src/utils/privateRoute.js:36]: Awesome bug tackle!

**Review by mxhs — APPROVED** [review]

File: src/store/actions/index.js (hunk 2 above)
Comment (mattcanavan) [reply src/store/actions/index.js:27]: I didn't think twice about removing this. I'll add it back in!

File: src/utils/privateRoute.js (hunk 3 above)
Comment (mattcanavan) [reply src/utils/privateRoute.js:36]: i just pushed a new commit that changed this file. is solved an issue where state was being changed on an unmounted component.

**Comment by mattcanavan** [comment]
TEST these changes on your local machine. After successful login the dashboard is not being rendered hmmm

**Review by CPower1248 — APPROVED** [review]
Thx for changing that reference, looks great

**Pull request merged**